src/test/java/bio/terra/janitor/db/JanitorDaoTest.java

```diff
@@ -0,0 +1,51 @@
+package bio.terra.janitor.db;
+
+import static bio.terra.janitor.app.common.TestUtils.*;
+import static bio.terra.janitor.common.ResourceType.GOOGLE_PROJECT;
+
+import bio.terra.generated.model.CloudResourceUid;
+import bio.terra.janitor.app.Main;
+import bio.terra.janitor.app.configuration.JanitorJdbcConfiguration;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@Tag("unit")
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = Main.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class JanitorDaoTest {
+  @Autowired JanitorJdbcConfiguration jdbcConfiguration;
+  @Autowired JanitorDao janitorDao;
+
+  private NamedParameterJdbcTemplate jdbcTemplate;
+
+  @BeforeEach
+  public void setup() {
+    jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+  }
+
+  @Test
+  public void createTrackedResource() throws Exception {
+    CloudResourceUid cloudResourceUid = newGoogleProjectResourceUid();
+    janitorDao.createResource(cloudResourceUid, DEFAULT_LABELS, CREATION, EXPIRATION);
+
+    assertCreateResultMatch(
+        cloudResourceUid,
+        GOOGLE_PROJECT,
+        Optional.of(CREATION),
+        Optional.of(EXPIRATION),
+        TIME_TO_LIVE_MINUTE,
+        jdbcTemplate,
+        DEFAULT_LABELS);
```

Comment:
ditto, odd that GOOGLE_PROJECT and TIME_TO_LIVE_MINUTE are implicit some of the time. I'd rather see this test be more explicit here, instead of relying on TestUtils to do everything.

Reply:
Refactor a little bit. Now in DaoTest, we will have assertCreateResultMatch in DaoTest, and it will call:
`assertCreateResultMatch(cloudResourceUid, GOOGLE_PROJECT, CREATION, EXPIRATION, DEFAULT_LABELS);`

I think the GOOGLE_PROJECT still make sense because the method above saying `CloudResourceUid cloudResourceUid = newGoogleProjectResourceUid();` so that is a google_project. Wdyt?

Reply:
I think GOOGLE_PROJECT is still confusing. take it as an argument, or rename to DEFAULT_ and include that in the function. Or just inline the cloudResourceUid creation, it's not that long.

Reply:
sorry, I've been misreading it as a string, not as the ResourceType. Maybe don't fully import it so that we have `ResourceType.GOOGLE_PROJECT`.

Reply:
The resource type is not parameter `janitorDao.createResource`, the logic to get it happens inside the Dao, and it is done by the visitor. 
And the resource type is not part of cloudResourceUid creation. 
In the test here, I pass GOOGLE_PROJECT to assert method is because I know the CloudResourceUid is a google project(and it doesn't require GOOGLE_PROJECT as an input). How can I build the connection here between `newGoogleProjectResourceUid` with   `GOOGLE_PROJECT`? One way I can come up with is call visitor.visit(cloudResourceUid) to gee the type?

Reply:
ah I see what you mean, do you want me to put 
```
cloudResourceUid = CloudResourceUid()
        .googleProjectUid(new GoogleProjectUid().projectId(UUID.randomUUID().toString()));
```

In the test here, not to use the static method in TestUtils?

Reply:
yeah, though `projectId("my-project")` is probably clearer than using a uuid. Sorry about the GOOGLE_PROJECT, was confused by the enum.

Reply:
I use the random UUID because each test run will insert one record into DB(DB is not getting reset) and I don't want those interrupt each other. 
Inlined the cloudResourceUid creation now..